### PR TITLE
[JUJU-1475] Added lxd constraints tests

### PR DESCRIPTION
**Pull request by anvial**
Added lxd tests to constraints suite. 

## Checklist

 - [ ] Comments answer the question of why design decisions were made

## QA steps

```sh
cd tests
./main.sh -v -p lxd constraints
```

## Documentation changes

None

## Bug reference

None

### CODE REVIEW

File: tests/suites/constraints/constraints.sh [hunk 1]
```diff
@@ -15,17 +43,17 @@ run_constraints_aws() {
   wait_for_machine_agent_status "1" "started"
   wait_for_machine_agent_status "2" "started"
 
-	echo "Ensure machine 0 has 16G root disk"
+  echo "Ensure machine 0 has 16G root disk"
   machine0_hardware=$(juju machines --format json | jq -r '.["machines"]["0"]["hardware"]')
-  machine0_rootdisk=$(echo "$machine0_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /root-disk/){print $i}}}')
-  check_ge "${machine0_rootdisk}" "root-disk=16384M"
+  machine0_cores=$(echo "$machine0_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /root-disk/){print $i}}}')
```
Comment (wallyworld) [inline tests/suites/constraints/constraints.sh:48]: this rename is wrong; the constraints being checked are root disk size

File: tests/suites/constraints/constraints.sh [hunk 2]
```diff
@@ -6,7 +35,6 @@ run_constraints_aws() {
 
   ensure "constraints-aws" "${file}"
 
-	echo "Deploy 3 machines with different constraints"
```
Comment (wallyworld) [inline tests/suites/constraints/constraints.sh:9]: no need to remove this

File: tests/suites/constraints/constraints.sh [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+
+  ensure "constraints-lxd" "${file}"
+
+  juju add-machine --constraints "cores=2"
+  juju add-machine --constraints "cores=2 mem=2G"
+
+  wait_for_machine_agent_status "0" "started"
+  wait_for_machine_agent_status "1" "started"
+
+	echo "Ensure machine 0 has 2 cores"
```
Comment (wallyworld) [inline tests/suites/constraints/constraints.sh:15]: white space

File: tests/suites/constraints/constraints.sh [hunk 4]
```diff
@@ -1,3 +1,32 @@
...
+  machine0_hardware=$(juju machines --format json | jq -r '.["machines"]["0"]["hardware"]')
+  machine0_cores=$(echo "$machine0_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /cores/){print $i}}}')
+  check_ge "${machine0_cores}" "cores=2"
+
+	echo "Ensure machine 1 has 2 cores and 2G memory"
```
Comment (wallyworld) [inline tests/suites/constraints/constraints.sh:20]: white space

File: tests/suites/constraints/constraints.sh [hunk 5]
```diff
@@ -1,3 +1,32 @@
...
+  machine1_hardware=$(juju machines --format json | jq -r '.["machines"]["1"]["hardware"]')
+  machine1_cores=$(echo "$machine1_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /cores/){print $i}}}')
+  machine1_mem=$(echo "$machine1_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /mem/){print $i}}}')
+  check_ge "${machine1_cores}" "cores=2"
+  check_ge "${machine1_mem}" "mem=2048M"
```
Comment (wallyworld) [inline tests/suites/constraints/constraints.sh:25]: for lxd, the memory allocation is <=
same with cores

**Review by wallyworld — COMMENTED** [review]
A few small things

File: tests/suites/constraints/constraints.sh (hunk 5 above)
Comment (anvial) [reply tests/suites/constraints/constraints.sh:25]: yep, I used refactofing tool (from Goland), that not correctly did its job. I'll fix it manually.

File: tests/suites/constraints/constraints.sh (hunk 2 above)
Comment (anvial) [reply tests/suites/constraints/constraints.sh:9]: Reverted

**Review by wallyworld — APPROVED** [review]

**Pull request merged**